src/main/java/uk/gov/hmcts/reform/translate/service/DictionaryService.java

```diff
@@ -50,4 +57,48 @@ public Map<String, String> getDictionaryContents() {
             throw new RoleMissingException(MANAGE_TRANSLATIONS_ROLE);
         }
     }
+
+
+    public void putDictionary(final Dictionary dictionaryRequest) {
+
+        val isManageTranslationRole = securityUtils.hasRole(MANAGE_TRANSLATIONS_ROLE);
+        val currentUserId = securityUtils.getUserInfo().getUid();
+        dictionaryRequest.getTranslations().entrySet()
+            .stream()
+            .forEach(englishPhrase -> processEachPhrase(englishPhrase, currentUserId, isManageTranslationRole));
+    }
+
+    private void processEachPhrase(Map.Entry<String, String> currentPhrase, String currentUserId,
+                                   boolean isManageTranslationRole) {
+
+        val result = dictionaryRepository.findByEnglishPhrase(currentPhrase.getKey());
+        if (result.isPresent()) {
+            updatePhrase(currentPhrase, currentUserId, result.get(), isManageTranslationRole);
```

Comment:
```suggestion
        if (existingPhrase.isPresent() && isManageTranslationRole) {
            updatePhrase(currentPhrase, currentUserId, result.get());
```
We could add here to make clearer when it is required, rather than in the `updatePhrase()` implementation. Since we'll never do anything in `updatePhrase()` when `isManageTranslationRole` is false.